docs/reference/api/connectivity/networksocket/DNS.md

```diff
@@ -0,0 +1,90 @@
+## DNS Resolver
+
+The DNS resolver provides an interface to do DNS host name resolutions.
+
+### Usage
```

Comment:
This section feels a lot like a tutorial. Is there any way to rephrase the numbered steps to give more background reference material than a step-by-step guide?

Reply:
Added a new chapter about asynchronous interface.